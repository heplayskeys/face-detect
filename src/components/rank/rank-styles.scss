$breakpoint-tablet: 768px;

.rank-container {
	width: 30%;

	@media screen and (max-width: $breakpoint-tablet) {
		width: 90%;
	}
}
