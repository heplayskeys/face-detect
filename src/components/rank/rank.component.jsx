import React from 'react';
import { connect } from 'react-redux';
import './rank-styles.scss';

const Rank = ({ entries }) => {
	return (
		<div className='rank-container dib ba b--white-50 br3 pa3 bg-black-10'>
			<div className='white f4'>{`You have contributed ${entries} images to face-grabber.`}</div>
		</div>
	);
};

const mapStateToProps = state => ({
	entries: state.setUser.activeUser.entries
});

export default connect(mapStateToProps)(Rank);
