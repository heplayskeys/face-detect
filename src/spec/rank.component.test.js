import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Rank } from '../components/rank/rank.component';

let wrapper;

describe('<Rank />', () => {
	beforeEach(() => {
		const mockProps = {
			entries: 10
		};
		wrapper = shallow(<Rank {...mockProps} />);
	});

	it('renders rank component with mock props', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
