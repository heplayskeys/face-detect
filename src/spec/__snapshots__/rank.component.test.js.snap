// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rank /> renders rank component with mock props 1`] = `
<div
  className="rank-container dib ba b--white-50 br3 pa3 bg-black-10"
>
  <div
    className="white f4"
  >
    You have contributed 10 images to face-grabber. [object Promise]
  </div>
</div>
`;
